packages/cli/src/babel-stuff.ts

```diff
@@ -0,0 +1,119 @@
+import * as babel from "@babel/core";
+import path from "path";
+import crypto from "crypto";
+import resolveFrom from "resolve-from";
+
+function normalizeOptions(config: {
```

Comment:
huh, I assume this is somewhat replicating the logic within Babel itself? is the caching worth the added complexity and having to sync stuff like this with future changes in Babel? this is used both by production builds but also by the internal hook in node and thus during tests etc, right? the latter is a compelling argument for this

Reply:
Yeah, it is replicating logic within Babel. I'm not super worried about the cost of syncing this with future changes in Babel given that the file this is from hasn't changed in like three years(https://github.com/babel/babel/blob/main/packages/babel-core/src/transformation/normalize-opts.js). I'm mostly concerned about whether this actually yields significant performance improvements and what the disk cost of it is.

> this is used both by production builds but also by the internal hook in node

That's the plan

> thus during tests

This isn't really true at least for Jest because they have their own require implementation which ignores require hooks

Reply:
> This isn't really true at least for Jest because they have their own require implementation which ignores require hooks

Good to know that those 2 ain't fully compatible.

> I'm mostly concerned about whether this actually yields significant performance improvements and what the disk cost of it is.

Right, let me know when u have some numbers - would be interesting to see the results of this experiment.